game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/OpponentSelector.java

```diff
@@ -0,0 +1,260 @@
+package games.strategy.triplea.odds.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.triplea.delegate.Matches;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.triplea.util.Tuple;
+
+/**
+ * Utility class with methods to pick players for the battle calculator.
+ *
+ * <p>Throughout the class the term "neutral player" is used, this denotes a player who is neither
+ * at war nor allied with another player.
+ */
+public class OpponentSelector {
+
+  /** Utility class, no instances please. */
```

Comment:
Minor, can replace the constructor and comment with class level 'lombok' annotation: `@UtilityClass`